Zilon.Core/Zilon.Bot.Players/BotActorTaskSourceBase.cs

```diff
@@ -74,17 +80,27 @@ public IActorTask[] GetActorTasks(IActor actor)
 
                 if (actorTask == null)
                 {
-                    return System.Array.Empty<IActorTask>();
+                    return Task.FromResult<IActorTask>(new IdleTask(actor, 1));
                 }
 
-                return new[] { actorTask };
+                return Task.FromResult(actorTask);
             }
             else
             {
                 _actorStrategies.Remove(actor);
             }
 
-            return System.Array.Empty<IActorTask>();
+            throw new System.Exception();
+        }
+
+        public void ProcessTaskComplete(IActorTask actorTask)
+        {
+            
+        }
+
+        public void ProcessTaskExecuted(IActorTask actorTask)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add a nested comment explaining why this method is empty, throw a 'NotSupportedException' or complete the implementation.](https://app.codacy.com/manual/kreghek/Zilon_Roguelike/pullRequest?prid=5544638)